auth.php

```diff
@@ -104,6 +104,19 @@ public function __construct() {
         $this->defaultidp = auth_saml2_get_default_idp();
     }
 
+    public static function saml2_authproc_filters_hook()
+    {
+        if (function_exists('get_plugins_with_function')) {
+            $callbacks = get_plugins_with_function('extend_auth_saml2_proc', 'lib.php');
+            foreach ($callbacks as $type => $plugins) {
+                foreach ($plugins as $plugin => $pluginfunction) {
+                    $authprocfilters = $pluginfunction();
```

Comment:
I think you want to keep appending more items onto the array, and also start the array with an explicit empty array

Reply:
updated with a default filter.